dotcom-rendering/src/components/TextBlockComponent.tsx

```diff
@@ -259,7 +259,7 @@ const buildElementTree =
 					if (
 						dropCappedSentence &&
 						node.textContent.startsWith(
```

Comment:
My understanding is that [`textContent`](https://developer.mozilla.org/en-US/docs/Web/API/Node/textContent) is already stripped of HTML tags.

Reply:
Saying that, drop cap can only be 1 or 2 characters long, so probably best to only compare the first two characters?

Reply:
I think in this case the node is broken up because of the link and the entire node is 6 characters long in this example ("That "), so the startsWith comparison against the entire text fails (html tags or no html tags). I think. I would be happy to change to 1 if the drop cap is always 1 char. 

But yeah no idea why it was 10 in the first place and what cases it might break if changed..

Reply:
[A drop cap can be 1 or 2 characters](https://github.com/guardian/dotcom-rendering/blob/d67be90cf0c2abdb968fd4b30945bd9093470025/dotcom-rendering/src/components/TextBlockComponent.tsx#L48-L74), so we’d need to check two at least.

Reply:
Pairing on this, we think that what this code is trying to achieve is actually check that this node is the child of the first child of the document fragment. This way we need not do any string comparison and make the logic less brittle in the case of improperly nested paragraphs[^1]

```ts
node.parentNode === node.parentNode?.parentNode
```

Reply:
Thanks for the notes Max! and for pairing 🙏

Reply:
| Before      | After      |
| ----------- | ---------- |
| <img width="795" alt="Screenshot 2023-10-13 at 17 14 31" src="https://github.com/guardian/dotcom-rendering/assets/1229808/c76d0c39-b6f2-4cc4-b2a9-bff78bc573b0"> | <img width="912" alt="Screenshot 2023-10-13 at 17 14 54" src="https://github.com/guardian/dotcom-rendering/assets/1229808/61264218-b85e-41a1-84bc-e161bdd5fa5e"> |